### DISCUSSION TITLE
Custom authentication using Okta for Hubs Cloud

### DISCUSSION THREAD

**Discussion by pflugs30** [mozilla/hubs#5451] [Hubs Cloud Questions] [answered]
I'm working on a project for a client who is testing Hubs Cloud Enterprise in AWS.  They would like to impose a requirement stating that all internal employees may access their Hubs Cloud instance as long as the Users authenticate through Okta, their OAuth provider.  We created an Okta application and client configuration which we tested using [this code](https://github.com/okta-samples/okta-react-sample).  Now we're trying to integrate the Okta authentication into a custom Hubs client.

I've been looking at this problem for several days, and I'm struggling to know where to start.  I don't have a lot of experience in React, having worked mostly with smaller React projects.  I'm trying to determine where to integrate Okta into the Hubs design, and I'm looking for ideas.

The Okta React sample project above uses a `SecureRoute` provided by [`@okta/okta-react`](https://www.npmjs.com/package/@okta/okta-react), but I'm not sure where or whether I could do the same in Hubs.  I also thought about adding some custom code into the Signin component itself.  The thought would be to require the User to first sign in via Okta's redirect loop and then signin again with the Hubs magic link (not efficient but acceptable to the client's project team).

I'm looking for any thoughts or guidance here, something to point me in a decent direction.  Any feedback?  Thank you!

**Comment by rawnsley** [comment, marked as answer]
We have a very similar requirement to integrate our OAuth provider with Hubs Cloud. We are using OryHydra for OAuth, but it should all be the same.

There has been some work towards this goal on the backend: https://github.com/mozilla/reticulum/pull/426. However, it is not currently in active development. In case you are not already aware, customising your own reticulum service isn't supported, but it is possible and you might be able to get help from the Discord community.

This topic has been discussed on Discord and I recommend searching there for more context. Here is an example: https://discord.com/channels/498741086295031808/570772852866023446/785651288082874398

**Reply by pflugs30** [reply]
Hello, @rawnsley!  Thanks for the response.  I'm reading through the issue you linked above.  Also, I tried to access the topic you linked in the Discord channel, but I believe it's within a private channel (#dev-extensibility).  I'm trying to figure out how to gain access to it now.  Thanks for the nudge!
